project.go

```diff
@@ -0,0 +1,392 @@
+		return err
+	}
+	err = c.setCNames(apps, client)
+	if err != nil {
+		appNames := make([]string, len(apps))
+		for i, app := range apps {
+			appNames[i] = app["name"]
+		}
+		deleteApps(appNames, client)
+		return fmt.Errorf("failed to configure project %q: %s", c.name, err)
+	}
+	fmt.Fprintf(ctx.Stdout, "successfully created the project %q!\n", c.name)
+	if gitRepo := apps[0]["repository_url"]; gitRepo != "" {
+		fmt.Fprintf(ctx.Stdout, "Git repository: %s\n", gitRepo)
+	}
+	return nil
+}
+
+func (c *projectCreate) Flags() *gnuflag.FlagSet {
+	if c.fs == nil {
+		c.fs = gnuflag.NewFlagSet("project-create", gnuflag.ExitOnError)
+		c.fs.StringVar(&c.name, "name", "", "name of the project")
+		c.fs.StringVar(&c.name, "n", "", "name of the project")
+		c.fs.StringVar(&c.platform, "platform", "", "platform of the project")
+		c.fs.StringVar(&c.platform, "l", "", "platform of the project")
+		c.fs.StringVar(&c.team, "team", "", "team that owns the project")
+		c.fs.StringVar(&c.team, "t", "", "team that owns the project")
+		c.fs.StringVar(&c.plan, "plan", "", "plan to use for the project")
+		c.fs.StringVar(&c.plan, "p", "", "plan to use for the project")
+		c.fs.Var(&c.envs, "envs", "comma-separated list of environments to use (defaults to dev,qa,stage,production)")
+		c.fs.Var(&c.envs, "e", "comma-separated list of environments to use (defaults to env,qa,stage,production)")
+		c.envs.Set("dev,qa,stage,production")
+	}
+	return c.fs
+}
+
+func (c *projectCreate) createApps(envs []Environment, client *cmd.Client) ([]map[string]string, error) {
+	createdApps := make([]map[string]string, 0, len(envs))
+	appNames := make([]string, 0, len(envs))
+	for _, env := range envs {
+		appName := fmt.Sprintf("%s-%s", c.name, env.Name)
+		app, err := createApp(client, createAppOptions{
+			name:     appName,
+			plan:     c.plan,
+			platform: c.platform,
+			pool:     env.poolName(),
+			team:     c.team,
+		})
+		if err != nil {
+			deleteApps(appNames, client)
+			return nil, fmt.Errorf("failed to create the project in env %q: %s", env.Name, err)
+		}
+		app["name"] = appName
+		app["dnsSuffix"] = env.DNSSuffix
+		createdApps = append(createdApps, app)
+		appNames = append(appNames, appName)
+	}
+	return createdApps, nil
+}
+
+func (c *projectCreate) setCNames(apps []map[string]string, client *cmd.Client) error {
+	for _, app := range apps {
+		reqURL, err := cmd.GetURL(fmt.Sprintf("/apps/%s/cname", app["name"]))
+		if err != nil {
+			return err
+		}
+		cname := fmt.Sprintf("%s.%s", c.name, app["dnsSuffix"])
+		v := make(url.Values)
+		v.Set("cname", cname)
+		req, err := http.NewRequest("POST", reqURL, strings.NewReader(v.Encode()))
+		if err != nil {
+			return err
+		}
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		resp.Body.Close()
+	}
+	return nil
+}
+
+func (c *projectCreate) filterEnvironments(envs []Environment, names []string) []Environment {
```

Comment:
Considering a slice is returned, I vote for keeping the name `filterEnvironments`.